src/components/PostsList/PostsList.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import './postsList.css';
+import posts from '../../api/posts';
+import users from '../../api/users';
+import comments from '../../api/comments';
+
+import Post from '../Post/Post';
+import Comments from '../Comments/Comments';
+
+function getPostsWithUsers(posts, users, comments) {
+  return (
+    posts.map(post =>
+    <Post post={post} user={users.find(user => post.userId === user.id)}>
+    <Comments
+      commentsArr={comments.filter((comment) => (
+        comment.postId === post.id
+      ))}
+    ></Comments>
+    </Post>
+    )
+  );
+};
+
+const PostsList = () => (
+  <section className="posts">
+    {getPostsWithUsers(posts, users, comments)}
```

Comment:
якщо ти використовуєш цю функцію тільки тут, то карще брати данні з замикання. Зараз ти створюєш shadow variable
```suggestion
    {getPostsWithUsers()}
```

Reply:
Согласен)